ibis/sql/tests/test_compiler.py

```diff
@@ -2417,3 +2418,50 @@ def test_agg_filter_with_alias():
   FROM t1
 )"""
     assert result == expected
+
+
+def test_table_drop_with_filter():
+    left = ibis.table(
+        [('a', 'int64'), ('b', 'string'), ('c', 'timestamp')],
+        name='t',
+    ).relabel({'c': 'C'})
+    left = left.filter(left.C == datetime.datetime(2018, 1, 1))
+    left = left.drop(['C'])
+    left = left.mutate(the_date=datetime.datetime(2018, 1, 1))
+
+    right = ibis.table([('b', 'string')], name='s')
+    joined = left.join(right, left.b == right.b)
+    joined = joined[left.a]
+    joined = joined.filter(joined.a < 1.0)
+    result = to_sql(joined)
+    # previously this was generating incorrect aliases due to not binding the
+    # self to expressions when calling projection:
```

Comment:
You've changed `table.projection` to `table.__getitem__` in the implementation. Does it mean that projection still buggish?

Reply:
Never mint, just read your description :)